src/components/modules/rectangleSelection.ts

```diff
@@ -96,33 +96,47 @@ export default class RectangleSelection extends Module {
    * Creating rect and hang handlers
    */
   public prepare(): void {
+    this.setReadOnly(this.config.readOnly);
+  }
+
+  /**
+   * Set read-only state
+   *
+   * @param {boolean} readOnlyEnabled
+   */
+  public setReadOnly(readOnlyEnabled: boolean) {
     const {Listeners} = this.Editor;
-    const {container} = this.genHTML();
 
-    Listeners.on(container, 'mousedown', (event: MouseEvent) => {
-      if (event.button !== this.MAIN_MOUSE_BUTTON) {
-        return;
-      }
-      this.startSelection(event.pageX, event.pageY);
-    }, false);
-
-    Listeners.on(document.body, 'mousemove', (event: MouseEvent) => {
-      this.changingRectangle(event);
-      this.scrollByZones(event.clientY);
-    }, false);
-
-    Listeners.on(document.body, 'mouseleave', () => {
-      this.clearSelection();
-      this.endSelection();
-    });
-
-    Listeners.on(window, 'scroll', (event) => {
-      this.changingRectangle(event);
-    }, false);
-
-    Listeners.on(document.body, 'mouseup', () => {
-      this.endSelection();
-    }, false);
+    if (readOnlyEnabled) {
+      Listeners.removeAll();
```

Comment:
it will remove all listeners, not only from this module

Reply:
I thought it was simpler to call `removeAll` considering each module's listener events aren't being added in a way that makes it easy to remove them. Should any listeners remain active in read-only mode?

https://github.com/hboylan/editor.js/blob/7b1cffe69516f33ec3313536fa8619fb55d26f4e/src/components/modules/dragNDrop.ts#L41-L53